modules/llvm-project/17.0.3/presubmit.yml

```diff
@@ -0,0 +1,39 @@
+matrix:
+  platform:
+  - debian10
+  - ubuntu2004
+tasks:
+  run_tests:
+    name: Run LLVM unit tests
+    platform: ${{ platform }}
+    test_flags:
+    - '--cxxopt=-std=c++17'
+    - '--host_cxxopt=-std=c++17'
+    test_targets:
+    - '@llvm-project//llvm/unittests:all'
+  run_tests_macos:
+    name: Run LLVM unit tests
+    platform: macos
+    test_flags:
+    - '--cxxopt=-std=c++17'
+    - '--host_cxxopt=-std=c++17'
+    - '--test_tmpdir=ci' # Avoid CI permissions error on macOS
```

Comment:
We're not sure why the `TEST_TMPDIR` (or `TMPDIR`) fails the permission tests on MacOS. Maybe the filesystem that is provided inside the sandbox does not quite propagate the permissions correctly? @meteorcloudy, any ideas? This works but seems like a CI setup issue, maybe?

Reply:
We get these errors when we do not override the temporary directory:
```[ RUN      ] FileSystemTest.permissions
Test Directory: /private/var/tmp/_bazel_buildkite/c7c6aa8230bc34b263b721a880582f7f/execroot/_main/_tmp/dad4610e8763dfa8e40c93b4abd0237e/file-system-test-a9c2d2
external/llvm-project~17.0.3/llvm/unittests/Support/Path.cpp:2293: Failure
Value of: CheckPermissions(fs::set_gid_on_exe)
  Actual: false
Expected: true
external/llvm-project~17.0.3/llvm/unittests/Support/Path.cpp:2308: Failure
Value of: CheckPermissions(fs::set_uid_on_exe | fs::set_gid_on_exe | fs::sticky_bit)
  Actual: false
Expected: true
external/llvm-project~17.0.3/llvm/unittests/Support/Path.cpp:2315: Failure
Value of: CheckPermissions(fs::all_read | fs::set_uid_on_exe | fs::set_gid_on_exe | fs::sticky_bit)
  Actual: false
Expected: true
external/llvm-project~17.0.3/llvm/unittests/Support/Path.cpp:2318: Failure
Value of: CheckPermissions(fs::all_perms)
  Actual: false
Expected: true
external/llvm-project~17.0.3/llvm/unittests/Support/Path.cpp:2323: Failure
Value of: CheckPermissions(fs::all_perms & ~fs::sticky_bit)
  Actual: false
Expected: true
[  FAILED  ] FileSystemTest.permissions (1 ms)```
Which just seem _odd_ that the permissions are not correctly set and read back.

The [test](https://github.com/llvm/llvm-project/blob/llvmorg-17.0.3/llvm/unittests/Support/Path.cpp#L2293) is pretty straight forward.

Reply:
Removing `--test_tmpdir=ci` from `presubmit.yml` should cause the macOS build to fail in CI with the error message above, unless CI setup was changed since last week.

Reply:
No idea what's going on, but the CI setup wasn't changed since last week. So let's keep the workaround.